src/foam/core/ClassInfoImpl.java

```diff
@@ -61,10 +61,14 @@ public ClassInfo getParent() {
     return parent_;
   }
 
-  public ClassInfo addProperty(PropertyInfo p) {
-    p.setClassInfo(this);
-    axioms.add(p);
-    axiomsByName_.put(p.getName(), p);
+  public ClassInfo addAxiom(Axiom a) {
+    // TODO: Should all axioms have setClassInfo? If not, create an interface
+    // that has setClassInfo and make PropertyInfo implement it.
+    if (a instanceof PropertyInfo) {
```

Comment:
Follow FOAM style guideline for spacing.

Reply:
Done